File: python/7 kyu/319.py
# https://www.codewars.com/kata/5a0b72484bebaefe60001867/train/python

# Passed

def distance(p1, p2):
    arr_length = len(p1)
    if arr_length == 0 or any(len(arr) != arr_length for arr in [p1, p2]):
        return -1
    
    diff_squares = [(p1[idx] - p2[idx]) ** 2 for idx in range(len(p1))]
    distance = sum(diff_squares) ** 0.5
    return distance

output = distance([2,1,3,1], [2,0,2,-1])
print(output)

File: nodejs/7 kyu/319.js
// https://www.codewars.com/kata/58e0bd6a79716b7fcf0013b1/train/javascript

// Passed

function getAges(sum, difference) {
  let result = null;
  if (sum >= 0 && difference >= 0) {
    let firstAge = (sum - difference) / 2;
    let secondAge = sum - firstAge;
    result = firstAge < 0 || secondAge < 0 ? null : [secondAge, firstAge];
  }

  return result;
}

const output = getAges(24, 4);
console.log(output);